HashRegistrarSimplified.sol

```diff
@@ -460,6 +464,43 @@ contract Registrar {
     }
 
     /**
+     * @dev Allows anyone to delete the owner and resolver records for a (subdomain of) a
+     *      name that is not currently owned in the registrar. If passing, eg, 'foo.bar.eth',
+     *      the owner and resolver fields on 'foo.bar.eth' and 'bar.eth' will all be cleared.
+     * @param labels A series of label hashes identifying the name to zero out, rooted at the
+     *        registrar's root. Must contain at least one element. For instance, to zero 
+     *        'foo.bar.eth' on a registrar that owns '.eth', pass an array containing
+     *        [sha3('foo'), sha3('bar')].
+     */
+    function eraseNode(bytes32[] labels) {
```

Comment:
This wouldn't work because `setOwner` would undo the work of `setSubnodeOwner`; the second time around the loop, your call to `setSubnodeOwner` would fail because you don't own the parent node. All the `setOwner` calls need to be executed after all the `setSubnodeOwner` calls.